packages/dotcom-ui-header/src/components/dropdown-navigation/dropdownNavigation.js

```
@@ -1,93 +1,211 @@
+  // so we shouldn't be showing the sticky header's dropdown when the
+  // sticky header is not visible, or show the regular header's dropdown
+  // when the sticky header is visible
+  const stickyHeader = document?.querySelector('.o-header--sticky')
+  const stickyHeaderObserver = new IntersectionObserver(
+    (changes) => {
+      for (let change of changes) {
+        if (
+          document.documentElement.clientWidth >= 780 ||
+          (document.documentElement.clientWidth < 780 && !change.isIntersecting)
+        ) {
+          const closestDropdownParent = document.activeElement.closest('.o-header__dropdown')
+          if (closestDropdownParent) {
+            const dropdownButton = closestDropdownParent.querySelector('.o-header__dropdown-button')
+            dropdownButton.removeAttribute('data-dropdown-button-active')
+            document.activeElement.blur()
+          }
+        }
+      }
+    },
+    { threshold: [0.01] }
+  )
+
+  stickyHeaderObserver.observe(stickyHeader)
+}
+
+const initProDropdown = async () => {
+  const proDropdowns = document.querySelectorAll('.o-header__dropdown.pro_navigation');
+  if (!proDropdowns) {
+    console.error('Navigation dropdown not found');
+    return;
+  }
+
+  try {
+    const links = await fetchLinks('https://pro-navigation.ft.com/api/links');
+    if (!validateLinks(links) || isEqual(PRO_NAVIGATION_DROPDOWN_DEFAULT_LIST, links)) {
+      return;
+    }
+
+    proDropdowns.forEach((dropdown) => updateLinksList(dropdown, links));
+  }
+  catch (error) {
+    console.error('Error updating dropdown navigation:', error);
+  }
+}
+
+const fetchLinks = async (url) => {
+  const response = await fetch(url);
+  if (!response.ok) {
+    throw new Error(`HTTP error! Status: ${response.status}`);
+  }
+  return response.json();
+};
+
+const validateLinks = (links) => {
+  if (!Array.isArray(links) || links.length === 0) {
+    console.error('Invalid links data format or empty links list');
```

theoretically this could throw an error rather than just log, as anyways the call is in a try catch.